src/controllers/comments.controller.js

```diff
@@ -140,5 +142,104 @@ class Comments {
     await util.setSuccess(200, 'Update is successfully', updateComment);
     return util.send(res);
   }
+
+  /**
+   *
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @returns {Object} return json object
+   * @memberof Comments
+   */
+  static async likeComment(req, res) {
+    const { id } = req.params;
+    const { username } = req.auth;
+    const comment = await commentsService.findOne(id);
+
+    if (!comment) {
+      util.setError(404, `Comment with id: ${id} does not exist.`);
+      return util.send(res);
+    }
+    let { likesCount, likeInfo } = comment;
+    const userHasLikedBefore = likeInfo.search(username);
+    if (userHasLikedBefore >= 0) {
+      util.setError(400, 'You liked this comment already');
+      return util.send(res);
+    }
+
+    likesCount += 1;
+    likeInfo = `${username}, ${likeInfo}`;
+    await commentsService.updateComment(id, { likesCount, likeInfo });
+
+    const formattedLikeInfo = Helper.formatLikeInfo(likeInfo);
+    util.setSuccess(201, { likesCount, formattedLikeInfo });
+    return util.send(res);
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @returns {Object} return json object
+   * @memberof Comments
+   */
+  static async updateLikeComment(req, res) {
+    const { id } = req.params;
+    const { username } = req.auth;
+    const comment = await commentsService.findOne(id);
+    if (!comment) {
+      util.setError(404, `Comment with id: ${id} does not exist.`);
+      return util.send(res);
+    }
+
+    let { likesCount, likeInfo } = comment;
+    const userHasLikedBefore = likeInfo.search(username);
+
+    if (userHasLikedBefore === -1) {
+      util.setError(400, 'You did not like this comment before');
+      return util.send(res);
+    }
+
+    likesCount -= 1;
+    likeInfo = likeInfo.replace(`${username}, `, '');
+    await commentsService.updateComment(id, { likesCount, likeInfo });
+
+    util.setSuccess(200, 'You unliked this comment successfully');
+    return util.send(res);
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @returns {Object} return json object
+   * @memberof Comments
+   */
+  static async getLikesComments(req, res) {
+    const { id } = req.params;
+    const { username } = req.auth;
+    const comment = await commentsService.findOne(id);
+
+    if (!comment) {
+      util.setError(404, `Comment with id: ${id} does not exist.`);
+      return util.send(res);
+    }
+    const { likesCount, likeInfo } = comment;
+
+    const userHasLikedBefore = likeInfo.search(username);
+
+    if (userHasLikedBefore === -1) {
+      util.setError(200, { data: { likesCount, likeInfo } });
```

Comment:
Better would be ```if (!userHasLikedBefore)```

Reply:
@Mnickii this search function returns -1 when no match was found. Which interns makes it a `truthy` value when being evaluated. 

When we find a match, a positive number is returned as well being `truthy` too.

What can we do in this instance?

Reply:
oh cool, i see likeInfo is a list of the users who've liked it. In this case then it's alright. @minega25

Reply:
I have tried using the not operator the condition is not being met